### PR TITLE
feat: add trainer configuration

### PR DESCRIPTION
**Pull request by MIchaelFU0403**
Add trainer configuration
## Description
Trainer is designed for training machine learning and ai models that support schedulers to make scheduling decision for each peer task. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: trainer/config/config.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import (
+	"errors"
+	"net"
+	"time"
+
+	"d7y.io/dragonfly/v2/pkg/net/ip"
+)
+
+type Config struct {
+	// Metrics configuration
```
Comment (zhoulizhen) [inline trainer/config/config.go:12]: // Metrics configuration.

File: trainer/config/config.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	Metrics MetricsConfig `yaml:"metrics" mapstructure:"metrics"`
+	// Network configuration.
+	Network NetworkConfig `yaml:"network" mapstructure:"network"`
+	// Server configuration.
+	Server ServerConfig `yaml:"server" mapstructure:"server"`
+	// Train configuration
```
Comment (zhoulizhen) [inline trainer/config/config.go:18]: // Train configuration.

File: trainer/config/config.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+type MetricsConfig struct {
+	// Enable metrics service.
+	Enable bool `yaml:"enable" mapstructure:"enable"`
+	// Metrics service address.
+	Addr string `yaml:"addr" mapstructure:"addr"`
+}
+
+// New default configuration
```
Comment (zhoulizhen) [inline trainer/config/config.go:86]: // New default configuration.

File: trainer/config/config.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+type GRPCConfig struct {
+	// AdvertiseIP is advertise ip.
+	AdvertiseIP net.IP `yaml:"advertiseIP" mapstructure:"advertiseIP"`
+
+	// ListenIP is listen ip, like: 0.0.0.0, 192.168.0.1.
+	ListenIP net.IP `yaml:"listenIP" mapstructure:"listenIP"`
+
+	// Port is listen port.
+	PortRange TCPListenPortRange `yaml:"port" mapstructure:"port"`
```
Comment (gaius-qi) [inline trainer/config/config.go:63]: Use `Port int`.

File: trainer/config/config.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+	// AccessKey is access key ID.
+	AccessKey string `yaml:"accessKey" mapstructure:"accessKey"`
+
+	// SecretKey is access key secret.
+	SecretKey string `yaml:"secretKey" mapstructure:"secretKey"`
+}
+
+type GRPCConfig struct {
+	// AdvertiseIP is advertise ip.
```
Comment (gaius-qi) [inline trainer/config/config.go:56]: Add `AdvertiseIP`.

File: trainer/config/config.go [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+type ServerConfig struct {
+	// GRPC server configuration.
+	GRPC GRPCConfig `yaml:"grpc" mapstructure:"grpc"`
+
+	// KeepAlive configuration.
+	KeepAlive KeepAliveConfig `yaml:"keepAlive" mapstructure:"keepAlive"`
+}
+
+type ObjectStorageConfig struct {
```
Comment (gaius-qi) [inline trainer/config/config.go:38]: Remove `ObjectStorageConfig`.

File: trainer/config/config.go [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+	// EnableIPv6 enables ipv6 for server.
+	EnableIPv6 bool `yaml:"enableIPv6" mapstructure:"enableIPv6"`
+}
+
+type ServerConfig struct {
+	// GRPC server configuration.
+	GRPC GRPCConfig `yaml:"grpc" mapstructure:"grpc"`
+
+	// KeepAlive configuration.
+	KeepAlive KeepAliveConfig `yaml:"keepAlive" mapstructure:"keepAlive"`
```
Comment (gaius-qi) [inline trainer/config/config.go:35]: Remove `KeepAlive`.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: trainer/config/constants.go [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+	// DefaultMetricsAddr is default address for metrics server.
+	DefaultMetricsAddr = ":8000"
+)
+
+var (
+	// DefaultCertIPAddresses is default ip addresses of certificate.
+	DefaultCertIPAddresses = []net.IP{ip.IPv4, ip.IPv6}
+
+	// DefaultCertDNSNames is default dns names of certificate.
+	DefaultCertDNSNames = []string{"dragonfly-scheduler", "dragonfly-scheduler.dragonfly-system.svc", "dragonfly-scheduler.dragonfly-system.svc.cluster.local"}
```
Comment (gaius-qi) [inline trainer/config/constants.go:28]: Rename `dragonfly-scheduler` to `dragonfly-trainer`.

File: trainer/config/testdata/trainer.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  certSpec:
+    dnsNames:
+      - foo
+    ipAddresses:
+      - 0.0.0.0
+    validityPeriod: 10m
+  
+manager:
+  addr: 127.0.0.1:65003
+
```
Comment (gaius-qi) [inline trainer/config/testdata/trainer.yaml:31]: Remove black line.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**